src/submit-modal/submit-modal.js

```diff
@@ -0,0 +1,299 @@
+              })
+    let periodsText = ''
+    if (periods.length === 1) {
+        periodsText = i18n.t('{{periodsCount}} period: {{periods}}', {
+            periodsCount: periods.length,
+            periods: periods[0],
+            nsSeparator: '-:-',
+        })
+    }
+    if (periods.length > 1 && periods.length <= 3) {
+        periodsText = i18n.t('{{periodsCount}} periods: {{periods}}', {
+            periodsCount: periods.length,
+            periods: periods.join(', '),
+            nsSeparator: '-:-',
+        })
+    }
+    if (periods.length > 3) {
+        periodsText = i18n.t(
+            '3+ periods: {{periods}}, and {{periodsCountLessThree}} more',
+            {
+                periodsCount: periods.length,
+                periods: periods.slice(0, 3).join(', '),
+                periodsCountLessThree: periods.length - 3,
+                nsSeparator: '-:-',
+            }
+        )
+    }
+    return `${name}, ${orgUnitText}, ${periodsText}`
+}
+
+const ConfirmModalContent = ({ exchange, requests, onClose, onSubmit }) => {
+    // this is very wordy, but did not have luck with i18nextscanner picking up from more compact versions...
+    let reportTranslationsString
+    const { baseUrl } = useConfig()
+    const reportCount = requests.length
+    const exchangeName = exchange?.displayName
+    const exchangeURL =
+        exchange?.target?.type === 'INTERNAL'
+            ? baseUrl
+            : exchange?.target?.api?.url
+    const exchangeHostName = exchangeURL?.split('//')[1] // remove protocol
+
+    if (exchange?.target?.type === 'INTERNAL') {
+        if (requests.length > 1) {
+            reportTranslationsString = i18n.t(
+                '{{reportCount}} reports to {{-exchangeName}} internally at {{-exchangeHostName}}',
+                {
+                    reportCount,
+                    exchangeName,
+                    exchangeHostName,
+                }
+            )
+        } else {
+            reportTranslationsString = i18n.t(
+                '{{reportCount}} report to {{-exchangeName}} internally at {{-exchangeHostName}}',
+                {
+                    reportCount,
+                    exchangeName,
+                    exchangeHostName,
+                }
+            )
+        }
+    } else {
+        if (requests.length > 1) {
+            reportTranslationsString = i18n.t(
+                '{{reportCount}} reports to {{-exchangeName}} at {{-exchangeHostName}}',
+                {
+                    reportCount,
+                    exchangeName,
+                    exchangeHostName,
+                }
+            )
+        } else {
+            reportTranslationsString = i18n.t(
+                '{{reportCount}} report to {{-exchangeName}} at {{-exchangeHostName}}',
+                {
+                    reportCount,
+                    exchangeName,
+                    exchangeHostName,
+                }
+            )
+        }
+    }
```

Comment:
Yes. Confirm. I actually asked the question in slack because this naive implementation like `length === 1 ? i18n.t('row') ? i18n.t('rows')`is what we're doing in a number of platform apps (including data entry app). I opened a ticket for fixing this for data entry app, and wanted to catch up with Jen about this (but haven't had time / felt that it was lower priority)

Reply:
OK, fair enough, let's tackle this at some later point in time.